main.py

```
@@ -1134,0 +1,1230 @@
+        """Process chunking for a single document."""
+        chunks = self.create_chunks(markdown_content, url)
+
+        # Create URL-specific chunk directory to prevent filename collisions
+        url_chunk_dir = f"{chunk_dir}/{filename.split('.')[-2]}"
+        self.save_chunks(chunks, url_chunk_dir, chunk_format)
+
+    def scrape_by_links_file(
+        self,
+        links_file: str,
+        output_dir: str,
+        save_chunks: bool = True,
```

Ruff (0.8.2)

Boolean-typed positional argument in function definition

(FBT001)